test/functional/bundleadd_v2/add-rc.bats

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env bats
+
+load "../testlib"
+
+setup() {
+
+	create_test_environment "$TEST_NAME"
+	create_bundle -n test-bundle1 -f /foo/test-file1 "$TEST_NAME"
+	create_bundle -L -n test-bundle2 -f /bar/test-file2 "$TEST_NAME"
+
+}
+
+teardown() {
+
+	destroy_test_environment "$TEST_NAME"
+
+}
+
+# ------------------------------------------
+# Good Cases (all good bundles)
+# ------------------------------------------
+
+@test "bundle-add output: adding one bundle" {
+
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS test-bundle1"
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		Starting download of remaining update content. This may take a while...
+
+		File /foo was not in a pack
+		.
+		Finishing download of update content...
+		Installing bundle(s) files...
+		.
+		Calling post-update helper scripts.
+		Successfully installed 1 bundle
+	EOM
+	)
+	assert_in_output "$expected_output"
+
+}
+
+@test "bundle-add output: adding multiple bundles" {
+
+	# for this test we need another installable bundle
+	create_bundle -n test-bundle3 -f /baz/test-file3 "$TEST_NAME"
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS test-bundle1 test-bundle3"
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		Starting download of remaining update content. This may take a while...
+
+		File /baz was not in a pack
+		.
+		Finishing download of update content...
+		Installing bundle(s) files...
+		..
+		Calling post-update helper scripts.
+		Successfully installed 2 bundles
+	EOM
+	)
+	assert_in_output "$expected_output"
+
+}
+
+# ------------------------------------------
+# Bad Cases (all bad bundles)
+# ------------------------------------------
+
+@test "bundle-add output: adding one bundle that is already added" {
+
+	run sudo sh -c "$SWUPD bundle-add $SWUPD_OPTS test-bundle2"
+	assert_status_is 18
```

Comment:
For the weird error codes such as this one let's add a comment explaining what it means, like
```bash
# 18 == EBUNDLEINSTALL
```
Just makes it easier for future developers who break tests because they change the exit status.

This should be done for the other instances of this in this file as well.

Reply:
There is a good argument for defining all of these in testlib.bash,, so it could say `assert_status_is $EBUNDLEINSTALL`, which would also make the test more readable.

Reply:
This is a great suggestion @icarus-sparry

Reply:
yes, I'll include those in the test library